src/inc/corprof.idl

```diff
@@ -2408,6 +2412,31 @@ interface ICorProfilerCallback8 : ICorProfilerCallback7
         [in] BOOL       fIsSafeToBlock);
 }
 
+[
+    object,
+    uuid(27583EC3-C8F5-482F-8052-194B8CE4705A),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback9 : ICorProfilerCallback8
+{
+    // This event is triggered whenever a dynamic method is garbage collected
+    // and subsequently unloaded.
+
+    HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
+
+    HRESULT EventPipeEventDelivered(
+        [in] REFGUID eventGuid,
+        [in] DWORD eventId,
+        [in] DWORD eventVersion,
+        [in] DWORD eventThreadId,
+        [in] LARGE_INTEGER eventTimestamp,
```

Comment:
I am suspect the timestamp is QueryPerformanceCounter.     This looks to be the only place we use timestamps in the profiling API, so we may be setting a precedent.   My preference is that we use something that is more universal (e.g. some UTC based tick like FILETIME)

If we don't want to do this we should add the 'GetFrequency' API to the interface but this is a poor substitute (clients need to know some sort of base, that is a pain to produce.   this is why I prefer just using FILETIME (the runtime can do the conversion.  

See __GetSystemTimeAsFileTime for a way to get the current time as a FILETIME.   This can a call to QueryPerformanceCounter gives you the information you need to convert from any QPC value to a FILETIME.   

@noahfalk